src/server/editorServices.ts

```diff
@@ -2542,4 +2552,11 @@ namespace ts.server {
             return false;
         }
     }
+
+    /* @internal */
+    export const enum ScriptInfoOrConfigKind { ScriptInfo, Config }
+    /* @internal */
+    export type ScriptInfoOrConfig =
+        | { readonly kind: ScriptInfoOrConfigKind.ScriptInfo, readonly info: ScriptInfo }

```

Comment:
Why do you need this? You can look determine if its sourceFile(config) by just checking 
```typescript
function isConfigFile(config: SourceFile | ScriptInfo) : config is SourceFile {
(config as SourceFile).kind !== undefined; 
}
```